libraries/default.rb

```diff
@@ -36,6 +36,13 @@ def install_package(name)
       end
     end
 
+    def pin_package(name, version)
```

Comment:
This is from the [apt cookbook](https://supermarket.chef.io/cookbooks/apt). It creates files in /etc/apt/preferences.d/ that restrict which versions of a package apt is allowed to install or update.

For example, we want to pin elasticsearch to the major version `2.4` so the `install-elasticsearch` recipe calls `pin_package("elasticsearch", "2.4.*")` which results in a file `/etc/apt/preferences.d/elasticsearch.pref` that contains:

    Package: elasticsearch
    Pin: version 2.4.*
    Pin-Priority: 700

Currently the latest release of elasticsearch is 4.6.1. Should 4.6.2 be released an update packages operation would update the package, whereas a 4.7.x release would not be allowed, i.e. we would only get patch-level automatic updates.